.gitignore

```diff
@@ -59,4 +59,11 @@ lib/
 # Jest
 #
 .jest/
-coverage/
\ No newline at end of file
+coverage/
+
+# Keystore
+*.keystore
+
+# Generated
+android/app/src/main/assets/
```

Comment:
I was getting the same error as james with the `Unable to load script from assets index.android.bundle`. I was poking around online and a good solution looked like to generate some files that went into this folder, the fix worked for me and seemed reasonable so I decided to keep it.

To actually answer your question, I think a good rule of thumb is that anything generated shouldn't be version controlled, since this folder was generated by some other stuff I added I put it in the `.gitignore`.